scvi/model/_multivi.py

```diff
@@ -0,0 +1,709 @@
+        ).format(
+            n_genes,
+            n_regions,
+            self.module.n_hidden,
+            self.module.n_latent,
+            n_layers_encoder,
+            n_layers_decoder,
+            dropout_rate,
+            latent_distribution,
+            deeply_inject_covariates,
+            gene_likelihood,
+        )
+        self.fully_paired = fully_paired
+        self.n_latent = n_latent
+        self.init_params_ = self._get_init_params(locals())
+        self.n_genes = n_genes
+
+    def train(
+        self,
+        max_epochs: int = 500,
+        lr: float = 1e-4,
+        use_gpu: Optional[Union[str, int, bool]] = None,
+        train_size: float = 0.9,
+        validation_size: Optional[float] = None,
+        batch_size: int = 128,
+        weight_decay: float = 1e-3,
+        eps: float = 1e-08,
+        early_stopping: bool = True,
+        save_best: bool = True,
+        check_val_every_n_epoch: Optional[int] = None,
+        n_steps_kl_warmup: Union[int, None] = None,
+        n_epochs_kl_warmup: Union[int, None] = 50,
+        adversarial_mixing: bool = True,
+        plan_kwargs: Optional[dict] = None,
+        **kwargs,
+    ):
+        """
+        Trains the model using amortized variational inference.
+
+        Parameters
+        ----------
+        max_epochs
+            Number of passes through the dataset.
+        lr
+            Learning rate for optimization.
+        use_gpu
+            Use default GPU if available (if None or True), or index of GPU to use (if int),
+            or name of GPU (if str), or use CPU (if False).
+        train_size
+            Size of training set in the range [0.0, 1.0].
+        validation_size
+            Size of the test set. If `None`, defaults to 1 - `train_size`. If
+            `train_size + validation_size < 1`, the remaining cells belong to a test set.
+        batch_size
+            Minibatch size to use during training.
+        weight_decay
+            weight decay regularization term for optimization
+        eps
+            Optimizer eps
+        early_stopping
+            Whether to perform early stopping with respect to the validation set.
+        save_best
+            Save the best model state with respect to the validation loss (default), or use the final
+            state in the training procedure
+        check_val_every_n_epoch
+            Check val every n train epochs. By default, val is not checked, unless `early_stopping` is `True`.
+            If so, val is checked every epoch.
+        n_steps_kl_warmup
+            Number of training steps (minibatches) to scale weight on KL divergences from 0 to 1.
+            Only activated when `n_epochs_kl_warmup` is set to None. If `None`, defaults
+            to `floor(0.75 * adata.n_obs)`.
+        n_epochs_kl_warmup
+            Number of epochs to scale weight on KL divergences from 0 to 1.
+            Overrides `n_steps_kl_warmup` when both are not `None`.
```

Comment:
Overrides `n_steps_kl_warmup` when `n_epochs_kl_warmup` is not `None`? Looks like this needs to be changed in trainingplans.py as well